### PR TITLE
Vendor gardener/gardener@v1.42.3

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area open-source
/kind enhancement
/platform aws

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
The following dependency is updated:
- github.com/gardener/gardener: v1.40.2 -> v1.42.3
```

### CODE REVIEW

**Comment by timebertt** [comment]
You can drop 442df3f2d94e34aee2801e0d7daa72bb5cedd152 in favor of vendoring a g/g version including https://github.com/gardener/gardener/pull/5588

File: go.mod [hunk 1]
```diff
@@ -9,29 +9,29 @@ require (
...
-	golang.org/x/tools v0.1.7
-	k8s.io/api v0.22.2
-	k8s.io/apiextensions-apiserver v0.22.2
-	k8s.io/apimachinery v0.22.2
-	k8s.io/apiserver v0.22.2
+	golang.org/x/tools v0.1.9
+	k8s.io/api v0.23.3
+	k8s.io/apiextensions-apiserver v0.23.3
+	k8s.io/apimachinery v0.23.3
+	k8s.io/apiserver v0.23.3
```
Comment (ialidzhikov) [inline go.mod:26]: The replace directives have to be updated to `v0.23.3` as well.

Comment (acumino) [reply go.mod:26]: Please check 6c08652c471f45df2255be7ede134c7fbaa596d7

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Review by dkistner — APPROVED** [review]
/lgtm

**Pull request merged**